src/stores/CrudStore.js

```diff
@@ -0,0 +1,131 @@
+import {
+  decorate,
+  action
+} from "mobx";
+import _ from 'lodash';
+
+/**
+ * CurdStore implement common CRUD tasks
+ */
+class CrudStore {
+  
+  constructor(service){
+    this.service = service;
+  }
+
+  /**
+   * Get a Record
+   * @param {} id 
+   */
+  async get(id){
+    try {
+      return await this.service.get(id);
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  /**
+   * Get all records
+   * @param {*} pageNum 
+   * @param {*} pageSize 
+   * @param {*} search 
+   */
+  async getList(pageNum, pageSize, search) {
+    try {
+      return await this.service.getList(pageNum, pageSize, search);
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  /**
+   * Remove record by ID
+   * @param {*} id 
+   */
+  async remove(id){
+    try {
+      return await this.service.remove(id);
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  /**
+   * Add record to DB
+   * @param {*} obj 
+   */  
+  async add(obj) {
+    try {
+      return await this.service.add(obj);
+    } catch (err) {
```

Comment:
The store shouldn't be catching the errors.
The validation errors are handled by the components for example.

Reply:
removed stores and also changed strategy on add/remove relations.
all components should be refactored on new issues
cc: @msantang78